src/header/common/range.rs

```diff
@@ -0,0 +1,185 @@
+use std::fmt::{self, Display};
+use std::str::FromStr;
+
+use header::{Header, HeaderFormat, RangeUnit};
+use header::parsing;
+
+/// The `Range` header defined by HTTP/1.0.
+///
+/// > The "Range" header field on a GET request modifies the method
+/// > semantics to request transfer of only one or more subranges of the
+/// > selected representation data, rather than the entire selected
+/// > representation data.
+///
+/// > Range = byte-ranges-specifier / other-ranges-specifier
+/// > other-ranges-specifier = other-range-unit "=" other-range-set
+/// > other-range-set = 1*VCHAR
+///
+/// # Example values
+/// * `bytes=1000-`
+/// * `bytes=-2000`
+/// * `bytes=0-1,30-40`
+/// * `custom_unit=0-123`
+///
+/// Spec: https://tools.ietf.org/html/rfc7233#section-3.1
+#[derive(PartialEq, Clone, Debug)]
+pub struct Range {
+    unit: RangeUnit,
+    ranges: Vec<RangeSpec>,
+}
+
+/// Each 'Range' header can contain one or more RangeSpecs.
+/// Each RangeSpec defines a range of units to fetch
+#[derive(PartialEq, Clone, Debug)]
+pub enum RangeSpec {
+    /// Get all bytes between x and y ("x-y")
+    FromTo(u64, u64),
+    /// Get all bytes starting from x ("x-")
+    AllFrom(u64),
+    /// Get last x bytes ("-x")
+    Last(u64)
+}
+
+impl Range {
+    /// Get the most common byte range header ("bytes=from-to")
+    pub fn simple(from: u64, to: u64) -> Range {
```

Comment:
Yeah, I wasn't convinced with `simple` neither, but that was the best I could come up with at the time. I'll change it to bytes()